test/terraform/module-pinning.bats

```diff
@@ -1,16 +1,22 @@
 load 'lib'
 
 function setup() {
-:
+  TMPFILE="$(mktemp /tmp/terraform-modules-XXXXXXXXXXX.txt)"
 }
 
 function teardown() {
+  #rm -f $TMPFILE
 :
 }
 
 @test "check if terraform modules are properly pinned" {
   skip_unless_terraform
-  run bash -c "cat *.tf | json2hcl -reverse | jq -r '..|select(.source?)|.source' | sort -u | sed 's/^.*?ref=//' | grep -Ev '^(tags/[0-9]+\\.[0-9]+.*|)$$'"
-  log "$output"
-  [ $status -ne 0 ]
+  grep -Eo '^\s*source\s*=\s*"(.*?)"' *.tf | cut -d'"' -f2 | sort -u > $TMPFILE
+  if [ -s $TMPFILE ]; then
+    # Verify the modules are pinned to `tags/x.y` or nothing at all (maybe using `version` parameter instead)
+    sed 's/^.*?ref=//' < $TMPFILE | grep -E '^(tags/[0-9]+\.[0-9]+.*|)$'
```

Comment:
why in `'^(tags/[0-9]+\.[0-9]+.*|)$'`, the first `.` escaped `\.`, but the second one not?

Reply:
The second one is not matching a `.`, but saying anything else (e.g `1.0` or `1.0-alpha` or 1.0.2`, etc)